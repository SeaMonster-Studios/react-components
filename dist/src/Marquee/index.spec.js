"use strict";

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _jestEmotion = require("jest-emotion");

var _emotion = require("emotion");

var emotion = _interopRequireWildcard(_emotion);

var _reactTestingLibrary = require("react-testing-library");

var _ = require("./");

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

//
afterEach(_reactTestingLibrary.cleanup);

expect.addSnapshotSerializer((0, _jestEmotion.createSerializer)(emotion));

describe("Marquee", function () {
  it("Renders", function () {
    var _renderSetup = renderSetup(),
        getByTestId = _renderSetup.getByTestId;

    var Marquee = getByTestId("component-content-marquee");

    expect(Marquee).toBeDefined();
    expect(Marquee).toMatchSnapshot();
  });
});

function renderSetup(overrides) {
  var props = Object.assign({
    image: "https://picsum.photos/1200/500",
    styles: ""
  }, overrides);

  var wrapper = (0, _reactTestingLibrary.render)(_react2.default.createElement(
    _.Marquee,
    props,
    _react2.default.createElement(
      "h2",
      null,
      "Try our newest flavor \u2014 ",
      _react2.default.createElement("br", null),
      _react2.default.createElement(
        "em",
        null,
        "banana by bananarama"
      )
    ),
    _react2.default.createElement(
      "button",
      null,
      "Get some"
    )
  ));

  return Object.assign({}, wrapper, {
    props: props
  });
}

//# sourceMappingURL=index.spec.js.map