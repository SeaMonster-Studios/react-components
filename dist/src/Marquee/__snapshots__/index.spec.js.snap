// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marquee Renders 1`] = `
.emotion-0 {
  background: linear-gradient(rgba(255,255,255,0),rgba(255,255,255,0)),url(https://picsum.photos/1200/500);
  background-size: cover;
  background-position: center center;
}

<div
  class="emotion-0"
  data-testid="component-content-marquee"
  style="min-height: 510px; padding: 45px 0px;"
>
  <h2>
    Try our newest flavor â€” 
    <br />
    <em>
      banana by bananarama
    </em>
  </h2>
  <button>
    Get some
  </button>
</div>
`;
